crates/sparrow-arrow/src/avro/avro_arrow.rs

```diff
@@ -13,8 +13,13 @@ use std::sync::Arc;
 /// represents a row and contains tuples with a string field name and the corresponding Avro Value,
 /// and returns a Result containing a vector of Arrow arrays (Vec<ArrayRef>) if the
 /// conversion is successful, or an ArrowError if an error occurs during conversion.
-pub fn avro_to_arrow(values: Vec<Vec<(String, Value)>>) -> Result<Vec<ArrayRef>, ArrowError> {
-    (0..values[0].len())
+pub fn avro_to_arrow(values: Vec<Vec<(String, Value)>>, should_include_publish_time: bool) -> Result<Vec<ArrayRef>, ArrowError> {
+    let upper_limit = if should_include_publish_time {
```

Comment:
For long term -- this logic seems error-prone. Rather than relying on offsets (and order of columns) and boolean flags, it seems like it would be better to use the schema and the presence/absence of a given field.